04_pig_latin/pig_latin.rb

```diff
@@ -1 +1,22 @@
-#write your code here
+def translate(sentence)
+  words = sentence.split(" ")
+  pig_latin_words = words.map { |word| translate_word(word) }
+  pig_latin_words.join(" ")
+end
+
+def translate_word(word)
+  capitalized = (word[0] == word[0].upcase)
+  word = word.downcase
+  if word =~ /\A[aeiou]/
+    result = word + "ay"
+  elsif word =~ /\A(qu)/
+    result = word[2..-1] + "quay"
+  elsif word =~ /\A([^aeiou]*qu)/
+    match = word.match(/\A([^aeiou]*qu)(.*)/)
+    result = match[2] + match[1] + "ay"
+  else
+    match = word.match(/\A([^aeiou]+)(.*)/)
+    result = match[2] + match[1] + "ay"
```

Comment:
Since you don't need to check whether or not match happens at this point, you could use #sub like so:

```suggestion
    result = word.sub(/\A([^aeiou]+)(.*)/, '\2\1ay')
```

`\1` and `\2` refer to the first and second capture groups, respectively.

You could go even further and stop the regex matching at the first vowel:

```suggestion
    first_vowel_index = word.index(/[aeiou]/)
    word[first_vowel_index..-1] + word[0...first_vowel_index] + "ay"
```